core/cont/src/TList.cxx

```diff
@@ -714,7 +714,7 @@ void TList::RecursiveRemove(TObject *obj)
    while (lnk) {
       next = lnk->Next();
       TObject *ob = lnk->GetObject();
-      if (ob->TestBit(kNotDeleted)) {
+      if (ob && ob->TestBit(kNotDeleted)) {
```

Comment:
> That's actually better.
Why?  This incurs a somewhat spurrious extra indentation ... personally I liked the original version better :)